nomad/fsm.go

```diff
@@ -473,10 +473,19 @@ func (n *nomadFSM) applyDrainUpdate(reqType structs.MessageType, buf []byte, ind
 			return fmt.Errorf("error looking up ACL token: %v", err)
 		}
 		if token == nil {
-			n.logger.Error("token did not exist during node drain update")
-			return fmt.Errorf("token did not exist during node drain update")
+			node, err := n.state.NodeBySecretID(nil, req.AuthToken)
```

Comment:
We check auth tokens in the FSM?! Well that's not going to work with Workload Identities, but obviously that is not a problem for this PR.

Reply:
The only reason we do this is to record the accessor ID in the drain events, so we can totally fix this in a later PR by passing the [`AuthenticatedIdentity.String()`](https://github.com/hashicorp/nomad/blob/v1.5.3/nomad/structs/structs.go#L510-L524) value in the arguments somewhere. I just didn't want to go messing with that in this PR